lib/nfc/protocols/mf_ultralight/mf_ultralight_poller_sync.c

```diff
@@ -251,6 +251,12 @@ static NfcCommand mf_ultralight_poller_read_callback(NfcGenericEvent event, void
         command = NfcCommandStop;
     } else if(mfu_event->type == MfUltralightPollerEventTypeAuthRequest) {
         mfu_event->data->auth_context.skip_auth = true;
+        if(mf_ultralight_support_feature(
+               mfu_poller->feature_set, MfUltralightFeatureSupportAuthenticate)) {
+            mfu_event->data->auth_context.skip_auth = false;
+            memset(
+                mfu_poller->auth_context.tdes_key.data, 0x00, MF_ULTRALIGHT_C_AUTH_DES_KEY_SIZE);
```

Comment:
I'm working on code that touches the MFUL sync poller, and I have to wonder why you're authenticating with all zero keys when that's not even the default key as shipped. Plus this causes authentication for non MFUL-C tags, which may use up the authentication limit if configured.